nautobot/core/tests/test_api.py

```diff
@@ -488,3 +488,33 @@ def test_related_by_invalid(self):
             response = self.client.post(url, data, format="json", **self.header)
         self.assertHttpStatus(response, status.HTTP_400_BAD_REQUEST)
         self.assertEqual(ipam_models.VLAN.objects.filter(name="Test VLAN 100").count(), 0)
+
+
+class APIOrderingTestCase(testing.APITestCase):
+    """
+    Testing integration with DRF's OrderingFilter.
+    """
+
+    @classmethod
+    def setUpTestData(cls):
+        cls.url = reverse("circuits-api:provider-list")
+
+    @override_settings(EXEMPT_VIEW_PERMISSIONS=["*"])
+    def test_ascending_sort(self):
+        """Tests that results are returned in the expected ascending order."""
+        response = self.client.get(f"{self.url}?ordering=name&limit=10", **self.header)
+        self.assertHttpStatus(response, 200)
+        self.assertEqual(
+            list(map(lambda p: p["name"], response.data["results"])),
+            list(Provider.objects.order_by("name").values_list("name", flat=True)[:10]),
+        )
+
+    @override_settings(EXEMPT_VIEW_PERMISSIONS=["*"])
+    def test_descending_sort(self):
+        """Tests that results are returned in the expected descending order."""
+        response = self.client.get(f"{self.url}?ordering=-name&limit=10", **self.header)
+        self.assertHttpStatus(response, 200)
+        self.assertEqual(
+            list(map(lambda p: p["name"], response.data["results"])),
+            list(Provider.objects.order_by("-name").values_list("name", flat=True)[:10]),
+        )
```

Comment:
Would be good to see this as a generic test in `APIViewTestCases.ListObjectsViewTestCase` that we can apply to each list endpoint.

Reply:
How would we identify which fields are sortable? In base we should at least test all direct attributes. I could see the need to add additional for FKs and their common sorting operation `name`.

Reply:
Does this filter support ordering by nested fields out-of-the-box? I didn't see anything to make me think so.

Could probably make a pretty decent first approximation by iterating over serializer.fields and skipping any that are RelatedField or SerializerMethodField instances.

Reply:
I suspect so when I was digging into the code: https://github.com/encode/django-rest-framework/blob/e08e606c82afd0d5ec82b2c58badec11a4ce825e/rest_framework/filters.py#L255

Reply:
☝️ Generic test base added, further discussion above in PR description